Dshop/apps/products_catalogue/tests/test_api_products.py

```diff
@@ -100,71 +98,122 @@ def assert_product_with_images(data):
 
 
 @pytest.mark.django_db
-def test_product_detail_with_images(authenticated_api_client, create_product_with_images):
+def test_product_detail_with_images(api_client_authed, create_product_with_images):
     url = reverse('products-api-detail', kwargs={'pk': create_product_with_images.id})
-    response = authenticated_api_client.get(url)
+    response = api_client_authed.get(url)
 
     assert response.status_code == 200
     assert_product_with_images(response.data)
 
 
 @pytest.mark.django_db
-def test_access_protected_resource(authenticated_api_client, create_active_product, create_inactive_product):
+def test_get_list_one(set_test_pagination_size, tv_product, inactive_product):
     url = reverse('products-api-list')
-    response = authenticated_api_client.get(url)
+    response = APIClient().get(url)
     assert response.status_code == 200
-
     results = response.data.get('results', [])
     assert len(results) == 1
-
     product_data = results[0]
-
+    assert response.data['count'] == 1
+    assert response.data['previous'] is None
+    assert response.data['next'] is None
     assert_active_object(product_data)
 
 
-def test_access_protected_resource_without_authentication(api_client):
-    url = reverse('products-api-list')
-    response = api_client.get(url)
+@pytest.mark.django_db
+def test_product_detail_404():
+    url = reverse('products-api-detail', kwargs={'pk': 6669})
+    response = APIClient().get(url)
+    assert response.status_code == 404
 
-    assert response.status_code == 401
+
+@pytest.fixture
+def set_test_pagination_size(settings, autouse=True):
+    settings.REST_FRAMEWORK['PAGE_SIZE'] = 5
 
 
+def test_pagination_size_in_tests():
+    assert settings.REST_FRAMEWORK['PAGE_SIZE'] == 5
+    # pagination size changed here: Dshop/Dshop/settings_tests.py
+
 @pytest.mark.django_db
-def test_product_detail(authenticated_api_client, create_active_product):
-    url = reverse('products-api-detail', kwargs={'pk': create_active_product.id})
-    response = authenticated_api_client.get(url)
+def test_product_list_empty(set_test_pagination_size):
+    response = APIClient().get(reverse("products-api-list"))
+    assert response.status_code == 200
+    assert response.data['results'] == []
+    assert response.data['count'] == 0
+    assert response.data['previous'] is None
+    assert response.data['next'] is None
 
+
+
+@pytest.mark.django_db
+def test_product_detail(tv_product):
+    url = reverse('products-api-detail', kwargs={'pk': tv_product.id})
+    response = APIClient().get(url)
```

Comment:
After this change, this test is only checking response code, not details. Please bring back ID and other details tests - the one removed. Or use a function to test product details. 
Testing just response code is not enough. 
Actually, this `test_product_detail_with_images` test, after changing to not authenticated user, would make the current one test not needed no? They both check the same thing.

Reply:
Assert added. `test_product_detail_with_images` doesn't need authentication any more.

I still think it may be beneficial to have tests for image related behavior  and general object retrieve.
Especially since separate tests are already there. They don't overlap.